admin/app/football/controllers/FrontsController.scala

```diff
@@ -20,24 +20,13 @@ import pa.Fixture
 import pa.LiveMatch
 
 
-class FrontsController(val wsClient: WSClient, val mode: Mode.Mode) extends Controller with ExecutionContexts with GetPaClient with Logging {
+class FrontsController(val wsClient: WSClient, val mode: Mode.Mode) extends Controller with ExecutionContexts with PaFootballClient with Logging {
   val SNAP_TYPE = "json.html"
   val SNAP_CSS = "football"
 
   def index = Action.async { implicit request =>
-    for {
-      competitions <- client.competitions.map(PA.filterCompetitions)
-      competitionTeams <- Future.traverse(competitions){ comp =>
-        client.teams(comp.competitionId, comp.startDate, comp.endDate).recover {
-          case e: PaClientErrorsException =>
-            // 'No data' is returned as an error by PA API. Therefore we ignore exception and return an empty list
-            log.error(s"Admin Football Fronts - PA Client error when requesting teams for competition ${comp}: ", e)
-            List()
-        }
-      }
-      allTeams = competitionTeams.flatten.distinct
-    } yield {
-      Cached(3600)(RevalidatableResult.Ok(views.html.football.fronts.index(competitions, allTeams)))
+    fetchCompetitionsAndTeams.map {
+      case (competitions, teams) => Cached(3600)(RevalidatableResult.Ok(views.html.football.fronts.index(competitions, teams)))
```

Comment:
Another case of magic cache lengths in this controller 😞  (not that you introduced this!)